src/ansiblelint/cli.py

```diff
@@ -196,10 +275,22 @@ def get_cli_parser() -> argparse.ArgumentParser:
     )
     parser.add_argument(
         "--write",
-        dest="write",
-        action="store_true",
-        help="Reformat YAML files to standardize spacing, quotes, etc. "
-        "Future versions will expand this option so it fixes more issues.",
+        dest="write_list",
+        # this is a tri-state argument that takes an optional comma separated list:
+        #   not provided, --write, --write=a,b,c
+        action=WriteArgAction,
+        help="Allow ansible-lint to reformat YAML files and run rule transforms "
+        "(Reformatting YAML files standardizes spacing, quotes, etc. "
+        "A rule transform can fix or simplify fixing issues identified by that rule). "
+        "You can limit the effective rule transforms (the 'write_list') by passing a "
+        "keywords 'all' or 'none' or a comma separated list of rule ids or rule tags. "
+        "YAML reformatting happens whenever '--write' or '--write=' is used. "
+        "'--write' and '--write=all' are equivalent: they allow all transforms to run. "
+        "The effective list of transforms comes from 'write_list' in the config file, "
+        "followed whatever '--write' args are provided on the commandline. "
+        "'--write=none' resets the list of transforms to allow reformatting YAML "
+        "without running any of the transforms (ie '--write=none,rule-id' will "
+        "ignore write_list in the config file and only run the rule-id transform).",
```

Comment:
This is too long for a help text. It presents as this wall of text:
```
  --write [WRITE_LIST]  Allow ansible-lint to reformat YAML files and run rule transforms (Reformatting YAML
                        files standardizes spacing, quotes, etc. A rule transform can fix or simplify fixing
                        issues identified by that rule). You can limit the effective rule transforms (the
                        'write_list') by passing a keywords 'all' or 'none' or a comma separated list of
                        rule ids or rule tags. YAML reformatting happens whenever '--write' or '--write=' is
                        used. '--write' and '--write=all' are equivalent: they allow all transforms to run.
                        The effective list of transforms comes from 'write_list' in the config file,
                        followed whatever '--write' args are provided on the commandline. '--write=none'
                        resets the list of transforms to allow reformatting YAML without running any of the
                        transforms (ie '--write=none,rule-id' will ignore write_list in the config file and
                        only run the rule-id transform).
```

I guess the next thing to do is figure out where this might go in the docs and then post a link from the help text.